hollow/src/main/java/com/netflix/hollow/core/index/HollowPrefixIndex.java

```diff
@@ -85,9 +86,10 @@ public HollowPrefixIndex(HollowReadStateEngine readStateEngine, String type, Str
     @SuppressWarnings("WeakerAccess")
     public HollowPrefixIndex(HollowReadStateEngine readStateEngine, String type, String fieldPath,
             int estimatedMaxStringDuplicates) {
+        Objects.requireNonNull(type, "Hollow Prefix Key Index creation failed because type was null");
+        Objects.requireNonNull(readStateEngine, "Hollow Prefix Key Index creation for type [" + type
```

Comment:
I'm glad that this changes from `IllegalArgumentException` to `NullPointerException`. Technically this could be a breaking change, but I think the risk is fine. For it to break a caller they'd have to have been specifically catching `IllegalArgumentException` to guard against this case. Probably worth calling out in the release notes and at least bumping the minor version instead of just the patch. Or maybe we do a major version bump and be good semver citizens.

I think the risk is low so don't feel strongly.

Reply:
Thanks for pointing that out. I had thought of it while making the code change but forgot when creating the PR and if you hadn't pointed it out I would've probably missed it. I will add it to the release notes and bump the minor version given low risk.